astropy/io/ascii/tdat.py

```diff
@@ -0,0 +1,815 @@
+            if hasattr(col, "meta"):
+                out_col.meta.update(col.meta)
+                if "index" in col.meta:
+                    if col.meta["index"] == "key":
+                        indices.insert(0, col.name)
+                    else:
+                        indices.append(col.name)
+        # Add indices, if specified
+        if len(indices) > 0:
+            for name in indices:
+                out.add_index(name)
+        return out
+
+
+class Tdat(core.BaseReader):
+    """TDAT format
+    See: https://heasarc.gsfc.nasa.gov/docs/software/dbdocs/tdat.html"
+
+    Example::
+      <HEADER>
+      # # and // are comments
+      table_name = example_table
+      table_description = "Example table"
+      #
+      # Table Parameters
+      #
+      field[id] = integer [meta.id] (key) // Unique ID
+      field[ra] = float:.4f_degree [pos.eq.ra] (index) // Right Ascension
+      field[name] = char12 [meta.id] // Name
+      #
+      # Virtual Parameters
+      #
+      table_author = Example et al.
+      #
+      # Data Format Specification
+      #
+      line[1] = id name ra
+      <DATA>
+      1|TargetOne|1.0|
+      2|TargetTwo|2.0|
+      <END>
+
+    The comments and keywords defined in the header, excepting common header
+    section titles and blank comments, are available via the output table
+    ``meta`` attribute::
+
+      >>> from astropy.io import ascii
+      >>> lines = ascii.tdat.make_example_data()
+      >>> data = ascii.read(lines, format='tdat')
+      >>> print(data.meta['comments'])
+      ['# and // are comments']
+      >>> for name, keyword in data.meta['keywords'].items():
+      ...   print(name, keyword)
+      table_name example_table
+      table_description Example table
+      table_author Example et al.
+
+
+    When writing to the TDAT format, the header will be auto-populated by
+    information in the Table, prioritizing information given in the Table.meta:
+
+    comments : list or string, (optional)
+    Table information which provide context. This information is
+    included in the header preceding all other lines and commented
+    out with #
+
+    keywords : dict, (optional, recommended)
+    Header keywords which will appear in the file as "name=value" lines.
+    Of particular importance are table_name, table_description,
+    and table_document_url.
+
+    If there is no Table.meta, this writer will attempt to automatically
+    generate the appropriate header information based on the table and
+    column properties and the recommendations for the TDAT format by HEASARC.
+
+    Example::
+    >>> from astropy.table import Table
+    >>> from io import StringIO
+    >>> t = Table(names=('reference_id', 'RA', 'Name'),
+    ...           data=[[1, 2, 3], [1.0, 2.0, 3.0], ['c', 'd', 'e']])
+    >>> t.meta['table_name'] = "astropy_table"
+    >>> out = StringIO()
+    >>> t.write(out, format="ascii.tdat")
+    >>> out.getvalue().splitlines()
```

Comment:
The splitlines output is not readable.
```suggestion
    >>> print(out.getvalue())
```